### PR TITLE
Improve README.md

### PR DESCRIPTION
**Pull request by Aversiste**
This PR reformats the README.md file to improve its usefulness. It now explains how to build the project, run the tests or compile the bindings.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,36 +28,84 @@ Supported Features
...
+
+We currently provide wrappers for two other languages: C++ and Python,
+respectively named libtuntp++ and pytuntap.  More instructions about them is
+provided in the [`bindings`](bindings/README.md) folder.
+
+The C++ library is built by default and can be disabled with the flag
+`ENABLE_CXX`.
+
+The Python library is disabled by default and requires both `ENABLE_CXX` and
+`ENABLE_PYTHON` to work.  You will also need Python 3.6 and Boost libraries.
```
Comment (chipot) [inline README.md]: Maybe we should not specify the minor release of python ? I think we are compatible with python3/2 in general, no ?

Comment (Aversiste) [reply README.md]: I just documented the existing build: the file `bindings/python/CMakeLists.txt` has a line with `find_package(... python36)`.

So I don't really know the level of compatibility of pytuntap as I am pretty sure I never tested it. You only can tell me about it 😄.

File: README.md [hunk 2]
```diff
@@ -24,36 +28,84 @@ Supported Features
...
+The following options can be tweaked:
+
+- `ENABLE_CXX`: Enable building of the C++ wrapper library libtuntap++;
+- `ENABLE_PYTHON`: Enable building of the Python wrapper library pytuntap;
+- `BUILD_TESTING`: Enable building of the regress tests;
+
+#### Other languages bindings
+
+We currently provide wrappers for two other languages: C++ and Python,
+respectively named libtuntp++ and pytuntap.  More instructions about them is
```
[the remark this answers is not in the document]
Comment (Aversiste) [reply README.md]: Fixed.

File: README.md [hunk 3]
```diff
@@ -24,36 +28,84 @@ Supported Features
...
+* cmake;
+* C and C++ compilers.
+
+#### Build
+
+This project is built with cmake:
+
+    $ mkdir build; cd build
+    $ cmake ../
+    $ make
```
[the remark this answers is not in the document]
Comment (Aversiste) [reply README.md]: I added two sub-sections under § Install, one to describe the effect of `BUILD_SHARED_LIBS` and the other to document the existence of `CMakeLists.txt.local`. Is it better ?

**Pull request merged**